File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/types/UseCaseObservable.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.types;

import io.reactivex.Observable;

public interface UseCaseObservable<T, Params> {

    Observable<T> runUseCase(Params... params);
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/types/UseCaseObservable.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.types

import io.reactivex.Observable

interface UseCaseObservable<T, Params> {

    fun runUseCase(vararg params: Params): Observable<T>
}
